Marlin/src/feature/temp_comp.h

```diff
@@ -0,0 +1,104 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2019 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+#include "../inc/MarlinConfig.h"
+
+/* Heaters */
+#define TEMP_COMP_MAX_BED_TEMP TEMP_COMPENSATION_MAX_BED_TEMP // To avoid heating errors we use this max temperature
+
+/* Probe calibration */
+#define TEMP_COMP_PROBE_START_TEMP          30    // Base measurement; z-offset == 0
+#define TEMP_COMP_PROBE_TEMP_RES             5    // Resolution in °C between measurements
+#define TEMP_COMP_PROBE_MEASUREMENTS        10    // Max. number of measurements to be stored (35 - 80°C)
+#define TEMP_COMP_PROBE_END_TEMP     (TEMP_COMP_PROBE_MEASUREMENTS*TEMP_COMP_PROBE_TEMP_RES+TEMP_COMP_PROBE_START_TEMP)
+
+/* Bed calibration */
+#define TEMP_COMP_BED_START_TEMP            60    // Base measurement; z-offset == 0
+#define TEMP_COMP_BED_TEMP_RES               5    // Resolution in °C between measurements
+#define TEMP_COMP_BED_MEASUREMENTS          10    // Max. number of measurements to be stored (65 - 110°C)
+#define TEMP_COMP_BED_END_TEMP       (TEMP_COMP_BED_MEASUREMENTS*TEMP_COMP_BED_TEMP_RES+TEMP_COMP_BED_START_TEMP)
+
+/* Extruder/hotend calibration */
+// Note: calibrating this cannot be automated, therefore we need to set it manually -> use less measurements
+#define TEMP_COMP_EXT_START_TEMP           180    // Base measurement; z-offset == 0
+#define TEMP_COMP_EXT_TEMP_RES              20    // Resolution in °C between measurements
+#define TEMP_COMP_EXT_MEASUREMENTS           5    // Max. number of measurements to be stored (200 - 280°C)
+#define TEMP_COMP_EXT_END_TEMP     (TEMP_COMP_EXT_MEASUREMENTS*TEMP_COMP_EXT_TEMP_RES+TEMP_COMP_EXT_START_TEMP)
```

Comment:
Should this nummerical constants be bracketed by
```cpp
#indef XXX
  #define XXX value
#endif 
```
to make them configurable - for the probably rare cases it could be needed?